sklearn/utils/_testing.py

```diff
@@ -605,6 +609,205 @@ def check_docstring_parameters(func, doc=None, ignore=None):
+        if False, checking is skipped for parameters.
+        Can only be set if `exclude_params` is None.
+
+    exclude_params : list of str or None, default=None
+        List of parameters to be excluded. If None, no parameters are excluded.
+        Can only be set if `include_params` is True.
+
+    include_attribs : list of str or bool, default=False
+        List of attributes to be included. If True, all attributes are included,
+        if False, checking is skipped for attributes.
+        Can only be set if `exclude_attribs` is None.
+
+    exclude_attribs : list of str or None, default=None
+        List of attributes to be excluded. If None, no attributes are excluded.
+        Can only be set if `include_attribs` is True.
+
+    include_returns : list of str or bool, default=False
+        List of returns to be included. If True, all returns are included,
+        if False, checking is skipped for returns.
+        Can only be set if `exclude_returns` is None.
+
+    exclude_returns : list of str or None, default=None
+        List of returns to be excluded. If None, no returns are excluded.
+        Can only be set if `include_returns` is True.
+
+    Examples
+    --------
+    >>> from sklearn.metrics import (mean_absolute_error, mean_squared_error,
+    ... median_absolute_error)
+    >>> from sklearn.utils.testing import assert_docstring_consistency
+    ... # doctest: +SKIP
+    >>> assert_docstring_consistency([mean_absolute_error, mean_squared_error],
+    ... include_params=['y_true', 'y_pred', 'sample_weight'])  # doctest: +SKIP
+    >>> assert_docstring_consistency([median_absolute_error, mean_squared_error],
+    ... include_params=True)  # doctest: +SKIP
+    """
+    from numpydoc import docscrape
+
+    Args = namedtuple("args", ["include", "exclude", "arg_name"])
+    section_dict = {
+        "Parameters": Args(include_params, exclude_params, "params"),
+        "Attributes": Args(include_attribs, exclude_attribs, "attribs"),
+        "Returns": Args(include_returns, exclude_returns, "returns"),
+    }
+    for section in list(section_dict):
+        args = section_dict[section]
+        if args.exclude and args.include is not True:
+            raise TypeError(
+                f"The 'exclude_{args.arg_name}' argument can be set only when the "
+                f"'include_{args.arg_name}' argument is True."
+            )
+        if args.include is False:
+            del section_dict[section]
+
+    doc_dict = dict()
+    for i, obj in enumerate(objects):
+        if isinstance(obj, docscrape.NumpyDocString):
+            doc_dict[f"Object {i}"] = obj
+        elif (
+            inspect.isdatadescriptor(obj)
+            or inspect.isfunction(obj)
+            or inspect.isclass(obj)
+        ):
+            doc = docscrape.NumpyDocString(inspect.getdoc(obj))
+            doc_dict[obj.__name__] = doc
+        else:
+            raise TypeError(
```

Comment:
a little test for this one would be nice

Reply:
done!